.github/workflows/buildFrontendLighthouseImage.yml

```diff
@@ -0,0 +1,78 @@
+name: Build frontend docker image with okta disabled
+
+on:
+  workflow_dispatch:
+  workflow_call:
+    inputs:
+      force_build:
+        description: Force build
+        required: false
+        default: false
+        type: boolean
+    outputs:
+      version:
+        description: docker image version
+        value: ${{ jobs.build_and_push_frontend_image.outputs.version }}
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}/frontend-lighthouse
+
+jobs:
+
+  frontend_changes:
+    with:
+      what_to_check: frontend
+    uses: ./.github/workflows/checkForChanges.yml
+
+  build_and_push_frontend_image:
+    # We want to build the image if there are changes in the frontend folder, or if we are forcing a build, or if we are on the main branch
+    if: needs.frontend_changes.outputs.has_changes == 'true' || inputs.force_build == true || github.ref == 'refs/heads/main'
+    needs: [frontend_changes]
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    outputs:
+      version: ${{ steps.set.outputs.version }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          tags: |
+            type=raw,value=latest,enable=${{ github.ref == format('refs/heads/{0}', github.event.repository.default_branch) }}
+            type=ref,event=branch
+            type=ref,event=pr
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v4
+        with:
+          context: ./
+          file: ./frontend/Dockerfile
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          build-args: |
```

Comment:
I keep coming back to these `build-args` changes, and it feels like a weird pattern to me. I don't often set environment variables as build arguments because I believe that should be left to set at runtime. 

Additional note: we should *_never_* set secrets at build time, please remove references to secrets from the Dockerfiles. These are _public_ images.

This lighthouse file should be removed and the variables should be overwritten at runtime.

Reply:
So `create-react-app` allows you to read environment variables to configure it before it builds the statics, any variables/secrets set in the code will be replaced with their values and then bundled inside the generated statics.

This already happens with the frontend app. if you go to simplereport.gov and look through the statics returned for react, you will find any variables passed in. There are no secrets in a single page app.

You also cant override runtime variables for statics, because there is nothing to dynamically change.

As far as `build-args` references to secrets in the docker image, for the frontend, its setting the environment variables on a build step image (`from ... as build`) and none of environment variables is set on the actual final image ([here](https://github.com/CDCgov/prime-simplereport/blob/refs/heads/zedd/5952-e2e-investigation/frontend/Dockerfile#L52)) that gets shipped to docker registry since only the build folder gets copied there.

Reply:
After chatting with Zedd, we will stick with his pattern here because it's a major part of the time reduction. If needed, we may add a dev server version for other development tasks at a future time.